### PR TITLE
Support attributes defined as a string

### PR DESCRIPTION
**Pull request by lostie**
Solves https://github.com/ministryofjustice/govuk_elements_form_builder/issues/68

This fix ensures that any element passed to the method is correctly converted to an array:

```ruby
irb(main):001:0> Array([])
=> []
irb(main):002:0> Array(1)
=> [1]
irb(main):003:0> Array(nil)
=> []
irb(main):004:0> Array('string')
=> ["string"]
irb(main):005:0> Array(:symbol)
=> [:symbol]
irb(main):006:0> Array(['string', :symbol])
=> ["string", :symbol]
```

### CODE REVIEW

**Comment by aliuk2012** [comment]
@robmckinnon @misaka @alan @ministryofjustice/ruby Could you review this PR please?

**Comment by SteveMarshall** [comment]
It would be nice to see more detail in the commit history.

In particular, commits should follow the [GOV.UK commit guidance](https://github.com/alphagov/styleguides/blob/master/git.md#commit-messages) and contain a descriptive message body. This might include a brief explanation of the source of the bug/why we care, and why the bug existed. A good (but perhaps slightly extreme?) example of a *great* commit message is [this, from @dcarley](https://github.com/alphagov/govuk-puppet/commit/63b36f93bf75a848e2125008aa1e880c5861cf46).

File: spec/lib/govuk_elements_form_builder/form_builder_spec.rb [hunk 1]
```diff
@@ -48,6 +48,19 @@ def size(method, size)
       ]
     end
 
+    it 'supports attributes defined as a string' do
+      output = builder.send method, 'name', class: 'custom-class'
```
Comment (SteveMarshall) [inline spec/lib/govuk_elements_form_builder/form_builder_spec.rb]: Is there a reason you've used `'name'` rather than `:name` as in the test directly above? It makes it look like two things are varying in this test when it's actually only the `class` (if I'm reading it right?).

File: spec/lib/govuk_elements_form_builder/form_builder_spec.rb [hunk 2]
```diff
@@ -48,6 +48,19 @@ def size(method, size)
...
+
+      expect_equal output, [
+        '<div class="form-group">',
+        '<label class="form-label" for="person_name">',
+        'Full name',
+        '</label>',
+        %'<#{element_for(method)} class="form-control custom-class" #{type_for(method, type)}name="person[name]" id="person_name" />',
+        '</div>'
+      ]
```
Comment (SteveMarshall) [inline spec/lib/govuk_elements_form_builder/form_builder_spec.rb]: It would be nice if the `custom-class` here weren't buried in the middle of a sea of markup. Maybe it'd be worth refactoring this out to an `expected_form_group` that accepts parameters, or something like that?

Comment (SteveMarshall) [inline spec/lib/govuk_elements_form_builder/form_builder_spec.rb]: (And, to be clear, I'd include refactoring out other uses of this same markup as part of that work, not just this instance.)

File: spec/lib/govuk_elements_form_builder/form_builder_spec.rb (hunk 1 above)
Comment (lostie) [inline spec/lib/govuk_elements_form_builder/form_builder_spec.rb]: The whole point of this PR is to **support** to pass the _attribute_ as a **string** (the issue is described [here](https://github.com/ministryofjustice/govuk_elements_form_builder/issues/68)), so that's exactly why I've defined this test.

The attribute `:name` (symbol) is varying in this test to `'name'` (string).

Comment (SteveMarshall) [inline spec/lib/govuk_elements_form_builder/form_builder_spec.rb]: OK, so I'd suggest removing the `, class: 'custom-class'` from this test, or having a second test for that behaviour.

**Comment by lostie** [comment]
@SteveMarshall ♻️

**Comment by robmckinnon** [comment]
Thanks for all the attention the gem's getting.

Agreed there's some duplication in spec expectations. This is intentional to make it easy for developers, including those not experienced in Ruby, to quickly see the output html markup for each spec. At the time many specs were copies of example markup from the [form section of the GOV.UK elements guide](http://govuk-elements.herokuapp.com/form-elements/).

The custom RSpec matcher code makes it harder to quickly determine what the expected html markup looks like. You have to think harder to compare markup expectations to the GOV.UK elements guide.

In PR https://github.com/ministryofjustice/govuk_elements_form_builder/pull/71 I've proposed a different way to remove the duplicated name input html expectations, which retains keeping most of the markup expectation in one place. See: https://github.com/ministryofjustice/govuk_elements_form_builder/pull/71 - What do you think?

**Comment by robmckinnon** [comment]
@SteveMarshall @slorek @aliuk2012 @lostie I find it hard to quickly visualise the html expectations after the spec refactoring in this pull request.

In an alternate pull request, I've had a go at removing spec duplication while retaining an html expectation that was easy to read and understand.

Can you please review this alternate pull request? It's here: https://github.com/ministryofjustice/govuk_elements_form_builder/pull/71

**Comment by aliuk2012** [comment]
I'm like and agree with the "Provide support to Form Builder to define attributes as a string" commit.  However, I have to agree with @robmckinnon about  spec refactoring, I'm find it pretty hard to follow.

I think @robmckinnon PR #71 might be a better option as a spec refactor. @lostie maybe it better to split this PR into two. I'd be happy to approve just the attributes as a string and then we can work on refactoring specs as a separate PR.

**Comment by aliuk2012** [comment]
@lostie have you been able to do some work on this PR? As I said previously I'm happy with the majority of the pr except for the refactoring of the tests.

**Comment by robmckinnon** [comment]
@aliuk2012 Maybe you could merge https://github.com/ministryofjustice/govuk_elements_form_builder/pull/71 first? Then the fix for attributes defined as a string could be made on top of that.
